components/style/values/specified/grid.rs

```diff
@@ -408,3 +408,150 @@ impl ToCss for TrackRepeatFirst {
 
 impl ComputedValueAsSpecified for TrackRepeatFirst {}
 no_viewport_percentage!(TrackRepeatFirst);
+
+/// The type of `repeat` function.
+///
+/// https://drafts.csswg.org/css-grid/#typedef-track-repeat
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum TrackRepeatType {
+    /// [`<track-repeat>`](https://drafts.csswg.org/css-grid/#typedef-track-repeat)
+    Auto,
+    /// [`<auto-repeat>`](https://drafts.csswg.org/css-grid/#typedef-auto-repeat)
+    Normal,
+    /// [`<fixed-repeat>`](https://drafts.csswg.org/css-grid/#typedef-fixed-repeat)
+    Fixed,
+}
+
+/// The structure corresponding to the various `repeat` functions.
+///
+/// https://drafts.csswg.org/css-grid/#typedef-track-repeat
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub struct TrackRepeat<L> {
+    /// The type of `repeat` (based on the values we encountered in parsing)
+    pub repeat_type: TrackRepeatType,
```

Comment:
But --- that's all something that matters when parsing, once parsed you don't need to store the fact that it's a particular type, no?